build.assets/notarize-binaries.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+set -e
+
+usage() {
+  echo "Notarizes MacOS binaries with xcrun."
+  echo "Usage: $(basename $0) [file ...]" 1>&2
+  exit 1
+}
+
+# Pulled from https://stackoverflow.com/a/26809278
+shell_array_to_json_array() {
+    echo -n '['
+    while [ $# -gt 0 ]; do
+        x=${1//\\/\\\\}
+        echo -n \"${x//\"/\\\"}\"
+        [ $# -gt 1 ] && echo -n ', '
+        shift
+    done
+    echo ']'
+}
+
+# This is largely pulled from `build-common.sh` but modified for this use case
+sign_and_notarize_binaries() {
+  local bundle_id="$1"
+  local targets=(${@:2})
+
+  local notarization_zip="teleport.zip"
+
+  local gondir=''
+  gondir="$(mktemp -d)"
+  #shellcheck disable=SC2064
+  trap "rm -fr '$gondir'" EXIT
+
+  # Gon configuration file needs a proper extension.
+  local goncfg="$gondir/gon.json"
+  # Note that xcrun stapler does not support stapling zip files.
+  # Instead, Apple wants you to staple notarization tickets to binaries then
+  # rebuild the zip file. That being said, Apple also does not support
+  # binaries directly. Rather, they must be archived and then notarized.
```

Comment:
What does "archived" mean in this context?

Reply:
This block is worded poorly. Updated